components/match2/wikis/heroes/legacy/match_maps_legacy.lua

```diff
@@ -0,0 +1,161 @@
+---
+-- @Liquipedia
+-- wiki=heroes
+-- page=Module:MatchMaps/Legacy
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Arguments = require('Module:Arguments')
+local Array = require('Module:Array')
+local Json = require('Module:Json')
+local Lua = require('Module:Lua')
+local Logic = require('Module:Logic')
+local PageVariableNamespace = require('Module:PageVariableNamespace')
+local Table = require('Module:Table')
+
+local Match = Lua.import('Module:Match')
+local MatchGroup = Lua.import('Module:MatchGroup')
+local MatchSubobjects = Lua.import('Module:Match/Subobjects')
+
+local OpponentLibraries = require('Module:OpponentLibraries')
+local Opponent = OpponentLibraries.Opponent
+
+local globalVars = PageVariableNamespace()
+
+local UNKNOWN_MAP = 'Unknown'
+local TBD = 'tbd'
+local DEFAULT_WIN = 'W'
+local FORFEIT = 'FF'
+local MAX_NUMBER_OF_OPPONENTS = 2
+
+local MatchMapsLegacy = {}
+
+-- invoked by Template:MatchMapsLua
+---@param frame Frame
+function MatchMapsLegacy.convertMap(frame)
+	local args = Arguments.getArgs(frame)
+	args.map = Table.extract(args, 'battleground') or UNKNOWN_MAP
+	args.winner = Table.extract(args, 'win')
+	return Json.stringify(args)
+end
+
+---@param args table
+function MatchMapsLegacy._handleLocation(args)
+	if args.location then
+		local matchStatus = Logic.readBool(args.finished) and 'was' or 'being'
+		local locationComment = 'Match ' .. matchStatus .. ' played in ' .. args.location
+		args.comment = args.comment and (args.comment .. '<br>' .. locationComment) or locationComment
+		args.location = nil
+	end
+end
+
+---@param args table
+function MatchMapsLegacy._handleMaps(args)
+	for key, jsonMap, mapIndex in Table.iter.pairsByPrefix(args, 'match') do
```

Comment:
sure the prefix for maps is `'match'`

Reply:
Si, inside bracketmatchsummary, maps key is `match`, same happened in lol

Reply:
just making sure :)